src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java

```diff
@@ -348,7 +356,14 @@ public void beginTree(DetailAST rootAST) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final String literal = ast.getText();
+        String literal = ast.getText();
+
+        // Manipulate text block contents to match identical string literal
+        if (TokenUtil.isOfType(ast, TokenTypes.TEXT_BLOCK_CONTENT)) {
+            literal =
+                CheckUtil.stripIndentAndInitialNewLineFromTextBlock(literal);
+            literal = QUOTE + literal + QUOTE;
+        }
```

Comment:
I have added this code to ensure that the compiled content of both the text block literal and traditional string literal match where they should.